contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -1609,3 +1609,24 @@ Example: "must be greater than \`request\`".
 be less than 256", "must be greater than or equal to 0".  Do not use words
 like "larger than", "bigger than", "more than", "higher than", etc.
 * When specifying numeric ranges, use inclusive ranges when possible.
+
+
+## Automatic Resource Allocation And Deallocation
+
+API objects often are union object contains the following:
+1. One or more fields identifying `Type` specific to API object.
+3. A union of all the fields that an object can represent.
```

Comment:
"A set of N fields, only one of which should be set at any given time - effectively a union"

?